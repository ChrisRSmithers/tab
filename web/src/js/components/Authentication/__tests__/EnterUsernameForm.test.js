/* eslint-env jest */

import React from 'react'
import {
  shallow
} from 'enzyme'
import toJson from 'enzyme-to-json'

const mockUserData = {
  id: 'abc-123'
}

describe('EnterUsernameForm tests', function () {
  it('renders without error', function () {
    const EnterUsernameForm = require('../EnterUsernameForm').default
    shallow(
      <EnterUsernameForm user={mockUserData} />
    )
  })

  it('matches expected snapshot', function () {
    const EnterUsernameForm = require('../EnterUsernameForm').default
    const wrapper = shallow(
      <EnterUsernameForm user={mockUserData} />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
