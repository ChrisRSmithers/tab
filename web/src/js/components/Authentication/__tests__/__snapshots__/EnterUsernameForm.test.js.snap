// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnterUsernameForm tests matches expected snapshot 1`] = `
<Paper
  circle={false}
  rounded={true}
  style={
    Object {
      "backgroundColor": "#FFF",
      "padding": 24,
    }
  }
  transitionEnabled={true}
  zDepth={1}
>
  <span
    style={
      Object {
        "fontSize": 20,
        "fontWeight": 500,
      }
    }
  >
    Choose a username
  </span>
  <UsernameField
    onKeyPress={[Function]}
    otherError={false}
    style={
      Object {
        "display": "block",
      }
    }
    usernameDuplicate={false}
  />
  <span
    style={
      Object {
        "display": "flex",
        "justifyContent": "flex-end",
        "marginTop": 30,
      }
    }
  >
    <RaisedButton
      disabled={false}
      fullWidth={false}
      label="NEXT"
      labelPosition="after"
      onClick={[Function]}
      primary={true}
      secondary={false}
    />
  </span>
</Paper>
`;
